PubNub/Network/Parsers/PNSubscribeParser.m

```diff
@@ -188,36 +177,27 @@ + (NSDictionary *)parsedServiceResponse:(id)response withData:(NSDictionary *)ad
         }
         processedResponse = @{@"events":feedEvents,@"timetoken":timeToken};
     }
-    
+
     return processedResponse;
 }
 
 
 #pragma mark - Events processing
 
-+ (NSMutableDictionary *)eventFromData:(id)data forChannel:(NSString *)channel
-                                 group:(NSString *)group
-              withAdditionalParserData:(NSDictionary *)additionalData {
++ (NSMutableDictionary *)eventFromData:(id)data withAdditionalParserData:(NSDictionary *)additionalData {
     
     NSMutableDictionary *event = [NSMutableDictionary new];
-    if ([channel length]) {
-        
-        event[(![group length] ? @"subscribedChannel": @"actualChannel")] = channel;
-    }
-    if ([group length]) {
-        
-        event[@"subscribedChannel"] = group;
-    }
-    
+    NSString *channel = data[kPNMessageChannelKey];
+    NSString *subscriptionMatch = data[kPNMessageSubscriptionMatchKey];
+    event[@"subscribedChannel"] = subscriptionMatch;
+    event[@"actualChannel"] = channel;
     BOOL isPresenceEvent = [PNChannel isPresenceObject:channel];
```

Comment:
@jzucker2 Because new subscribe API pass channel name directly into message payload, we can use only this check, but changed a bit to find out whether this is presence event or not. Here how it should look: _[PNChannel isPresenceObject:(subscriptionMatch?: channel)];_